### PR TITLE
Documentation: Update Database.ReadAsync description

### PR DESCRIPTION
**Pull request by SaurabhSharma-MSFT**
# Pull Request Template

## Description

Documentation: Update Database.ReadAsync description.
Fixes [30503](https://github.com/Azure/azure-sdk-for-net/issues/30503)

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 1]
```diff
@@ -32,7 +32,8 @@ public abstract class Database
         public abstract CosmosClient Client { get; }
 
         /// <summary>
-        /// Reads a <see cref="DatabaseProperties"/> from the Azure Cosmos service as an asynchronous operation.
+        /// Reads a <see cref="DatabaseProperties"/> from the Azure Cosmos service as an asynchronous operation. 
+        /// However, there is an implicit conversion from DatabaseResponse to DatabaseProperties
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs:36]: Why is this note needed? All `XXXXXProperties` across the SDK have implicit conversions from the `XXXXXResponse`, the code of the `XXXXXResponse` states it so.

For example: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.databaseresponse?view=azure-dotnet#operators

**Review by ealsur — CHANGES_REQUESTED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs (hunk 1 above)
Comment (SaurabhSharma-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: Makes sense.  Updated the file as per your recommendations.

**Review by FabianMeiswinkel — DISMISSED** [review]
Thx

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 2]
```diff
@@ -31,15 +31,18 @@ public abstract class Database
...
+        /// <summary>        
+        /// Reads a <see cref="DatabaseResponse"/> from the Azure Cosmos service as an asynchronous operation.
         /// </summary>
         /// <param name="requestOptions">(Optional) The options for the request.</param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
         /// <returns>
         /// A <see cref="Task"/> containing a <see cref="DatabaseResponse"/> which wraps a <see cref="DatabaseProperties"/> containing the read resource record.
         /// </returns>
         /// <exception>https://aka.ms/cosmosdb-dot-net-exceptions</exception>
+        <remarks>
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs:43]: Missing the `///`
```suggestion
///        <remarks>
```

**Review by ealsur — CHANGES_REQUESTED** [review]
remarks is not being correctly escaped

**Review by ealsur — APPROVED** [review]

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Linq/CosmosLinqExtensions.cs [hunk 3]
```diff
@@ -174,19 +174,19 @@ public static FeedIterator<T> ToFeedIterator<T>(this IQueryable<T> query)
         }
 
         /// <summary>
-        /// This extension method gets the FeedIterator from LINQ IQueryable to execute query asynchronously.
-        /// This will create the fresh new FeedIterator when called.
+        /// This extension method gets the StreamIterator from LINQ IQueryable to execute query asynchronously.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Linq/CosmosLinqExtensions.cs:177]: Can we undo these changes to this file? `StreamIterator` is not a used term. Let's just do the `Database.cs` file changes so we can merge it without extra signoffs?

**Review by ealsur — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**